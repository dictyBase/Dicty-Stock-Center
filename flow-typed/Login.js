declare module 'components/OauthSignInButton' {
    declare module.exports: any;
}

declare module 'utils/clientConfig' {
    declare module.exports: any;
}
