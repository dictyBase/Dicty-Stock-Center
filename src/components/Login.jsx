// @flow
import React, { Component } from 'react'
import OauthSignInButton from 'components/OauthSignInButton'
import clientConfig from 'utils/clientConfig'
import { Flex, Box } from 'rebass'
import { DictyHeader } from 'styles'

const getDefaultProviders = () => {
    let providers = []
    for (let name in clientConfig) {
        providers.push(name)
    }
    return providers
}

type Props = {
    providers: Array<string>
}

export default class Login extends Component<Props> {
    displayName = 'login display'

    static defaultProps = {
        providers: getDefaultProviders()
    }
    renderOauthButtons = () => {
        const { providers } = this.props
        return providers.map((provider, index) => {
            return (
                <OauthSignInButton
                    provider={ provider }
                    key={ index }
                    {...this.props}
                />
            )
        })
    }
    render() {
        return (
            <Flex justify="center">
                <Box w={ [1, 1 / 2, 1 / 3] }>
                    <DictyHeader>
                        <h1>Log in</h1>
                    </DictyHeader>
                    { this.renderOauthButtons() }
                </Box>
            </Flex>
        )
    }
}
