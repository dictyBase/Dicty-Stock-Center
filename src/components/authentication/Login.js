import React, { Component } from "react"
import { Login as LoginContainer } from "dicty-components-login"
import OauthSignHandler from "components/authentication/OauthSignHandler"
import oauthConfig from "utils/oauthConfig"
import ErrorNotification from "components/authentication/ErrorNotification"
import Grid from "@material-ui/core/Grid"

// list of buttons to display
const buttons = ["orcid", "google", "linkedin"]

/**
 * Component that displays all of the social login buttons with click handlers for each one
 */

class Login extends Component {
  handleClick = name => {
    const config = oauthConfig[name]
    let url = `${config.authorizationEndpoint}?client_id=${config.clientId}`
    url += `&scope=${config.scopes.join(config.scopeDelimiter)}`
    if (config.requiredUrlParams) {
      config.requiredUrlParams.forEach(element => {
        url += `&${element[0]}=${element[1]}`
      })
    }
    if (config.optionalUrlParams) {
      config.optionalUrlParams.forEach(element => {
        url += `&${element[0]}=${element[1]}`
      })
    }
    url += `&redirect_uri=${config.redirectUrl}`
    window.open(
      url,
      name,
      `width=${config.popupOptions.width},
                    height=${config.popupOptions.height}`,
    )
  }
  render() {
    const { state = {} } = this.props.location
    const { error } = state
    return (
      <Grid container justify="center">
        <Grid item xs={8}>
          <center>
            <h1>Log in</h1>
          </center>
          {error && <ErrorNotification error={error} />}
          <Grid container justify="center">
            <Grid item xs={1} />
            <Grid item xs={4}>
              <LoginContainer buttons={buttons} onClick={this.handleClick} />
              <OauthSignHandler />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    )
  }
}

export default Login
