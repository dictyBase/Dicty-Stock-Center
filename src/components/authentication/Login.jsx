// @flow
import React, { Component } from "react"
import { Login as LoginContainer } from "dicty-components-login"
import OauthSignHandler from "components/authentication/OauthSignHandler"
import oauthConfig from "utils/oauthConfig"
import ProtectedRouteNotification from "components/authentication/ProtectedRouteNotification"
import { Flex, Box } from "rebass"
import { DictyHeader } from "styles"

// list of buttons to display
const buttons = ["orcid", "google", "linkedin"]

// custom theme for styling the buttons
const theme = {
  overrides: {
    MuiButton: {
      // name of the stylesheet
      root: {
        // name of the rule
        borderRadius: 3,
        color: "white",
        width: "80%",
        justifyContent: "start",
        minHeight: "55px",
        marginBottom: "5px"
      }
    }
  }
}

type Props = {
  // Object passed by React-Router
  location: Object
}

/**
 * Component that displays all of the social login buttons with click handlers for each one
 */

class Login extends Component<Props> {
  handleClick = (name: string) => {
    const config = oauthConfig[name]
    let url = `${config.authorizationEndpoint}?client_id=${config.clientId}`
    url += `&scope=${config.scopes.join(config.scopeDelimiter)}`
    if (config.requiredUrlParams) {
      config.requiredUrlParams.forEach(element => {
        url += `&${element[0]}=${element[1]}`
      })
    }
    if (config.optionalUrlParams) {
      config.optionalUrlParams.forEach(element => {
        url += `&${element[0]}=${element[1]}`
      })
    }
    url += `&redirect_uri=${config.redirectUrl}`
    window.open(
      url,
      name,
      `width=${config.popupOptions.width},
                    height=${config.popupOptions.height}`
    )
  }
  render() {
    const { state = {} } = this.props.location
    const { error } = state
    return (
      <Flex justify="center">
        <Box w={[1, "60%", "40%"]}>
          <DictyHeader>
            <h1>Log in</h1>
          </DictyHeader>
          {error && <ProtectedRouteNotification error={error} />}
          <Flex justify="center">
            <Box w={"15%"} />
            <Box w={"85%"}>
              <LoginContainer
                buttons={buttons}
                theme={theme}
                onClick={this.handleClick}
              />
              <OauthSignHandler />
            </Box>
          </Flex>
        </Box>
      </Flex>
    )
  }
}

export default Login
