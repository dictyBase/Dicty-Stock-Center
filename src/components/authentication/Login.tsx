import React, { Component } from "react"
import { Login as LoginContainer } from "dicty-components-login"
import OauthSignHandler from "components/authentication/OauthSignHandler"
import oauthConfig from "utils/oauthConfig"
import Grid from "@material-ui/core/Grid"

// list of buttons to display
const buttons = ["orcid", "google", "linkedin"]

const formatURLParams = (params: Array<string>) => {
  let url = ""
  params.forEach(element => {
    url += `&${element[0]}=${element[1]}`
  })
  return url
}

/**
 * Component that displays all of the social login buttons with click handlers for each one
 */

class Login extends Component {
  handleClick = (name: string) => {
    const config = oauthConfig[name]
    let url = `${config.authorizationEndpoint}?client_id=${config.clientId}`
    url += `&scope=${config.scopes.join(config.scopeDelimiter)}`
    if (config.requiredUrlParams) {
      url += formatURLParams(config.requiredUrlParams)
    }
    if (config.optionalUrlParams) {
      url += formatURLParams(config.optionalUrlParams)
    }
    url += `&redirect_uri=${config.redirectUrl}`
    window.open(
      url,
      name,
      `width=${config.popupOptions.width},
                    height=${config.popupOptions.height}`,
    )
  }
  render() {
    return (
      <Grid container justify="center">
        <Grid item xs={8}>
          <div style={{ textAlign: "center" }}>
            <h1>Log in</h1>
          </div>
          <Grid container justify="center">
            <Grid item xs={1} />
            <Grid item xs={4}>
              <LoginContainer buttons={buttons} onClick={this.handleClick} />
              <OauthSignHandler />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    )
  }
}

export default Login
