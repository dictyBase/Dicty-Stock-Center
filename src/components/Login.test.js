import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Login from "./Login"

test("matching a snapshot of Login", () => {
  const component = renderer.create(<Login />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
