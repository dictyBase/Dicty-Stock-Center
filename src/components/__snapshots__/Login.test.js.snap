// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Login 1`] = `
<div
  className="sc-bxivhb bUZGRr sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
  is={null}
>
  <div
    className="sc-ifAKCX cFlEyZ sc-bdVaJa dbOKEe"
    is={null}
  >
    <div
      className="sc-TFwJa daqYjB"
    >
      <h1>
        Log in
      </h1>
    </div>
    <div
      className="sc-bxivhb bUZGRr sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa brzzaE"
        is={null}
      >
        <a
          className="large block facebook sc-cooIXK jhyMRn"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-facebook"
          />
            Sign in with 
          Facebook
        </a>
      </div>
    </div>
    <div
      className="sc-bxivhb bUZGRr sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa brzzaE"
        is={null}
      >
        <a
          className="large block google sc-cooIXK jhyMRn"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-google"
          />
            Sign in with 
          Google
        </a>
      </div>
    </div>
    <div
      className="sc-bxivhb bUZGRr sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa brzzaE"
        is={null}
      >
        <a
          className="large block linkedin sc-cooIXK jhyMRn"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-linkedin"
          />
            Sign in with 
          LinkedIn
        </a>
      </div>
    </div>
    <div
      className="sc-bxivhb bUZGRr sc-htpNat iZBBQa sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa brzzaE"
        is={null}
      >
        <a
          className="large block orcid sc-cooIXK jhyMRn"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-orcid"
          />
            Sign in with 
          ORCID
        </a>
      </div>
    </div>
  </div>
</div>
`;
